submarine-server/server-core/src/main/java/org/apache/submarine/server/SubmitterManager.java

```diff
@@ -30,44 +31,49 @@
 import java.net.URL;
 import java.net.URLClassLoader;
 import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.List;
-import java.util.concurrent.ConcurrentHashMap;
-import java.util.concurrent.ConcurrentMap;
+import java.util.Map;
 
 /**
- * Submitter Manager, load all the submitter plugin, configured by
- * {@link SubmarineConfiguration.ConfVars#SUBMARINE_SUBMITTERS} and related key.
+ * Submitter Manager is responsible for load {@link JobSubmitter} implements class.
  */
 public class SubmitterManager {
   private static final Logger LOG = LoggerFactory.getLogger(SubmitterManager.class);
 
-  private SubmarineConfiguration conf;
-  private ConcurrentMap<String, JobSubmitter> submitterMap = new ConcurrentHashMap<>();
+  private static final SubmitterManager INSTANCE = new SubmitterManager();
 
-  public SubmitterManager(SubmarineConfiguration conf) {
-    this.conf = conf;
-    loadSubmitters();
+  private final Map<String, JobSubmitterConfig> SUBMITTER_TABLE = new HashMap<>();
+
+  private JobSubmitter submitter;
+
+  {
```

Comment:
We'd better move the initialization logic to a constructor method.

Reply:
Using the initializing instance members to split the initialize logic and this approach can be used to share a block of code between multiple constructors. So that it can avoid some mistakes.
More info see https://docs.oracle.com/javase/tutorial/java/javaOO/initial.html